co-log/src/Colog/Message.hs

```diff
@@ -296,18 +303,49 @@ fmtRichMessageDefault RichMessage{..} = do
      <> thread
      <> msgText
 
-    showTime :: C.Time -> Text
-    showTime t =
-        square
-        $ toStrict
-        $ TB.toLazyText
-        $ C.builder_DmyHMS timePrecision datetimeFormat (C.timeToDatetime t)
-      where
-        timePrecision = C.SubsecondPrecisionFixed 3
-        datetimeFormat = C.DatetimeFormat (Just '-') (Just ' ') (Just ':')
-
-    showThreadId :: ThreadId -> Text
-    showThreadId = square . T.pack . show
+{- | Formats 'RichMessage' in the following way:
+
+@
+[Time] [SourceLocation] [ThreadId] \<Text message\>
+@
+
+__Examples:__
+
+@
+[03 05 2019 05:23:19.058] [Main.example#34] [ThreadId 11] app: First message...
+[03 05 2019 05:23:19.059] [Main.example#35] [ThreadId 11] app: Second message...
+@
+
+Practically, it formats a message as 'fmtRichMessageDefault' without the severity information.
+-}
+fmtSimpleRichMessageDefault :: MonadIO m => RichMessage m -> m Text
+fmtSimpleRichMessageDefault msg = fmtRichMessageCustom msg formatRichMessage
+   where
+    formatRichMessage :: Maybe ThreadId -> Maybe C.Time -> Message -> Text
+    formatRichMessage (maybe "" showThreadId -> thread) (maybe "" showTime -> time) Msg{..} =
+        time
+     <> showSourceLoc msgStack
+     <> thread
+     <> msgText
+
+fmtRichMessageCustom :: MonadIO m => RichMessage m -> (Maybe ThreadId -> Maybe C.Time -> Message -> Text) -> m Text
```

Comment:
This function is not modular enough. The problem is that with both `fmtRichMessageDefault` and `fmtSimpleRichMessageDefault` you give `RichMessage m` as an argument, where `fmtSimpleRichMessageDefault` should have the following type:

```haskell
fmtSimpleRichMessageDefault :: MonadIO m => RichMsg m SimpleMsg -> m Text
```

And then you can generalize type of this function to the following:

```haskell
fmtRichMessageCustomDefault
    :: MonadIO m 
    => RichMessage m msg 
    -> (Maybe ThreadId -> Maybe C.Time -> msg -> Text) 
    -> m Text
```

Reply:
I've implemented these changes. Trying to have `tutorials/Main.hs` to compile I've noticed that the various log actions (`logInfo` etc.) all accept a `Message` or `Msg`, but `SimpleMsg` is neither... should we try to make those functions more generic as well or am I chasing down a rabbit hole?

Reply:
Well, `SimpleMsg` doesn't have `Severity` that's why it makes sense why you cannot use functions like `logInfo`. However, we have function `logMsg` which works with a polymorphic message:

https://github.com/kowainik/co-log/blob/a91ad4bdcc3a805e567ae124f5c4b729d95932df/co-log/src/Colog/Monad.hs#L56

I think it's better to create a new example with the `WithLog env Text m` constraint, so you can run it with your new functions.

Reply:
> Well, SimpleMsg doesn't have Severity

*facepalming hard*. Sorry, I must have been asleep.

I'm trying with creating a new example and having a 
`withLog (cmap fmtSimpleRichMessageDefault) simpleExample`
inside the `app` function of the tutorial.    
Getting a bit lost with the type tetris, but I'll get there :muscle:

Reply:
Keep up the good work! 👍

Reply:
@chshersh thanks! I've pushed another diff:    
- `RichMessage` was both a type and a constructor and GHC was getting quite confused, so I renamed the constructor `RichMessagex` just to make it compile. Do you have any idea on how to solve this? or do you have a better name?     
- as you can see, the approach I took in `play-colog` is a bit different from what was there before, I tried using the `withLog` function for days without getting much done. I don't like my current implementation (having to explicitly create a `RichMsg m SimpleMsg` is pretty awkward) but couldn't find a better way. Really hoping for you enlightenment :yin_yang: